server/src/main/java/org/apache/druid/server/http/RulesResource.java

```diff
@@ -152,6 +155,30 @@ public Response getDatasourceRuleHistory(
     }
   }
 
+  @GET
+  @Path("/{dataSourceName}/intervals/{interval}/dropped")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ResourceFilters(RulesResourceFilter.class)
+  public Response isSpecificIntervalDroppedByRule(
+      @PathParam("dataSourceName") String dataSourceName,
+      @PathParam("interval") String interval
+  )
+  {
+    final List<Rule> rules = databaseRuleManager.getRulesWithDefault(dataSourceName);
+    final Interval theInterval = Intervals.of(interval.replace('_', '/'));
+    final DateTime now = DateTimes.nowUtc();
+    boolean dropped = false;
+    for (Rule rule : rules) {
+      if (rule.appliesTo(theInterval, now)) {
+        if (rule instanceof DropRule) {
+          dropped = true;
```

Comment:
This looks not correct. First of all, rules can be empty. This effectively makes any segments to not be loaded into historicals. Also, only the first matched rule is applied. Even if there is a matched drop rule, a segment can be loaded if any load rule matches before checking the drop rule.

Reply:
Oh you're right. Thanks. I missed the `break`.